sqlglot/dialects/dialect.py

```diff
@@ -1748,11 +1751,19 @@ def _builder(args: t.List) -> F:
 
         # This is done to avoid failing in the expression validator due to the arg count
         del args[2:]
-        return expr_type(
-            this=seq_get(args, 0),
-            expression=exp.JSONPath(expressions=segments),
-            only_json_types=arrow_req_json_type,
-        )
+        kwargs = {
+            "this": seq_get(args, 0),
+            "expression": exp.JSONPath(expressions=segments),
+        }
+
+        is_jsonb = issubclass(expr_type, (exp.JSONBExtract, exp.JSONBExtractScalar))
+        if not is_jsonb:
+            kwargs["only_json_types"] = arrow_req_json_type
```

Comment:
Why was this necessary? Not sure if this is correct, doesn't seem bad to just set `only_json_types` regardless, right?

Reply:
Ah, I see, I wasn't aware of that. It's fine then.